pkgs/by-name/wi/wivrn/package.nix

```diff
@@ -0,0 +1,146 @@
+{
+  config,
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchFromGitLab,
+  applyPatches,
+  autoAddDriverRunpath,
+  avahi,
+  boost,
+  cli11,
+  cmake,
+  cudaPackages ? { },
+  cudaSupport ? config.cudaSupport,
+  eigen,
+  ffmpeg,
+  freetype,
+  git,
+  glm,
+  glslang,
+  harfbuzz,
+  libdrm,
+  libGL,
+  libva,
+  libpulseaudio,
+  libX11,
+  libXrandr,
+  nix-update-script,
+  nlohmann_json,
+  onnxruntime,
+  openxr-loader,
+  pipewire,
+  pkg-config,
+  python3,
+  shaderc,
+  spdlog,
+  systemd,
+  udev,
+  vulkan-headers,
+  vulkan-loader,
+  vulkan-tools,
+  x264,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wivrn";
+  version = "0.18";
+
+  src = fetchFromGitHub {
+    owner = "meumeu";
+    repo = "wivrn";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-JFiDjx2iQfBDD1yRWHPx0UiIAqEk8+nzf967o/HXYqs=";
+  };
+
+  monado = applyPatches {
+    src = fetchFromGitLab {
+      domain = "gitlab.freedesktop.org";
+      owner = "monado";
+      repo = "monado";
+      rev = "dfc602288ab05131584a3f2be18031a13fccd061";
+      hash = "sha256-4HZs3cgqOWWpXQb5kfG513f7+znO0hJvAbj2rxrqmeI=";
+    };
+
+    patches = [
+      "${finalAttrs.src}/patches/monado/0001-c-multi-disable-dropping-of-old-frames.patch"
+      "${finalAttrs.src}/patches/monado/0002-ipc-server-Always-listen-to-stdin.patch"
+      "${finalAttrs.src}/patches/monado/0003-c-multi-Don-t-log-frame-time-diff.patch"
+    ];
+  };
+
+  strictDeps = true;
+
+  postUnpack = ''
+    # Let's make sure our monado source revision matches what is used by WiVRn upstream
+    ourMonadoRev="${finalAttrs.monado.src.rev}"
+    theirMonadoRev=$(grep -A1 "https://gitlab.freedesktop.org/monado/monado" ${finalAttrs.src.name}/CMakeLists.txt | tail -n1 | sed 's/.*GIT_TAG\s*//')
+    if [ ! "$theirMonadoRev" == "$ourMonadoRev" ]; then
+      echo "Our Monado source revision doesn't match CMakeLists.txt." >&2
+      echo "  theirs: $theirMonadoRev" >&2
+      echo "    ours: $ourMonadoRev" >&2
+      return 1
+    fi
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    git
+    pkg-config
+    python3
+  ] ++ lib.optionals cudaSupport [ autoAddDriverRunpath ];
+
+  buildInputs = [
+    avahi
+    boost
+    cli11
+    eigen
+    ffmpeg
+    freetype
+    glm
+    glslang
```

Comment:
You need to move `glslang` to `nativeBuildInputs` to fix the build failure.

`glslangValidator` is used during configure/build time by Monado so to facilitate cross compiling it needs to be a native build input.

```diff
diff --git a/pkgs/by-name/wi/wivrn/package.nix b/pkgs/by-name/wi/wivrn/package.nix
index 59a8a15d45e6..6c1241c5e2e4 100644
--- a/pkgs/by-name/wi/wivrn/package.nix
+++ b/pkgs/by-name/wi/wivrn/package.nix
@@ -85,6 +85,7 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [
     cmake
     git
+    glslang
     pkg-config
     python3
   ] ++ lib.optionals cudaSupport [ autoAddDriverRunpath ];
@@ -97,7 +98,6 @@ stdenv.mkDerivation (finalAttrs: {
     ffmpeg
     freetype
     glm
-    glslang
     harfbuzz
     libdrm
     libGL
```